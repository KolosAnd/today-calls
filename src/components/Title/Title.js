import styles from "./Title.module.css"
import React from "react";

export const Title = (props) => {
    return (
        <div className={styles.title}>
            <h2 className={`${styles.title__name} ${props.classes}`}>{props.text}</h2>
        </div>
    )
}

