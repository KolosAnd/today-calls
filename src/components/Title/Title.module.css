.title {
}
.title__name {
    color: #fff;
    margin: 0 0 25px;
    font-size: 25px;
    text-decoration: none;
    line-height: 25px;
}
.title .title__name .main-title {
     margin: 0;
 }
