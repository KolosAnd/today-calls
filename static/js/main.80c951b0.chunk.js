(this["webpackJsonptoday-calls-app"]=this["webpackJsonptoday-calls-app"]||[]).push([[0],[,,,function(e,t,l){e.exports={nextCall:"NextCall_nextCall__2hiNO",nextCall__wrap:"NextCall_nextCall__wrap__E-JmG",nextCall__title:"NextCall_nextCall__title__3uk50",nextCall__inputWrap:"NextCall_nextCall__inputWrap__3k8d4",nextCall__inputsWrap:"NextCall_nextCall__inputsWrap__2i2Go",nextCall__input:"NextCall_nextCall__input__3VO86",nextCall__inputName:"NextCall_nextCall__inputName__3dOlZ",nextCall__inputPhone:"NextCall_nextCall__inputPhone__37SD1",nextCall__inputTime:"NextCall_nextCall__inputTime__2rI6V"}},,,,function(e,t,l){e.exports={header:"Header_header__2vzj5",header__container:"Header_header__container__3VRL9",header__title:"Header_header__title__2pv23"}},,function(e,t,l){e.exports={title:"Title_title__38Rue",title__name:"Title_title__name__G7AnC","main-title":"Title_main-title__15ozn"}},function(e,t,l){},,function(e,t,l){e.exports={button:"Button_button__9OKbO"}},,,,,,function(e,t,l){},,function(e,t,l){},function(e,t,l){"use strict";l.r(t);var a=l(1),c=l.n(a),s=l(11),n=l.n(s),i=l(13),r=l(2),o=(l(18),l(4)),_=l(9),j=l.n(_),d=l(0),u=function(e){return Object(d.jsx)("div",{className:j.a.title,children:Object(d.jsx)("h2",{className:"".concat(j.a.title__name," ").concat(e.classes),children:e.text})})},b=l(12),m=l.n(b),x=function(e){return Object(d.jsx)("button",{onClick:e.click,type:e.type,className:"".concat(m.a.button," ").concat(e.classes),children:e.text})},O=(l(10),function(){return{addItemToStorage:Object(a.useCallback)((function(e,t,l,a,c){var s={id:Date.now(),name:e,phone:t,time:l,milisec:a,finish:c};localStorage.setItem("".concat(a),JSON.stringify(s))}),[]),getItemsFromStorage:Object(a.useCallback)((function(){var e=[],t=[];for(var l in localStorage)localStorage.hasOwnProperty(l)&&(e.push(localStorage.getItem(l)),e.forEach((function(e){t.push(JSON.parse(e))})));return t}),[]),deleteItemsFromStorage:Object(a.useCallback)((function(e){localStorage.removeItem(e)}),[])}}),h=function(e){var t=e.create,l=Object(a.useState)({name:"",phone:"",time:""}),c=Object(r.a)(l,2),s=c[0],n=c[1],i=Object(a.useState)(""),_=Object(r.a)(i,2),j=_[0],b=_[1],m=Object(a.useState)(""),h=Object(r.a)(m,2),p=h[0],v=h[1],N=Object(a.useState)(""),f=Object(r.a)(N,2),C=f[0],g=f[1],S=O().addItemToStorage;var k=/^[0-9\b]+$/;return Object(d.jsxs)("div",{className:"add-call",children:[Object(d.jsx)(u,{text:"Add call"}),Object(d.jsxs)("form",{action:"",children:[Object(d.jsxs)("div",{className:"add-call__inputs-wrap",children:[Object(d.jsx)("input",{type:"text",value:j,onChange:function(e){b(e.target.value),n(Object(o.a)(Object(o.a)({},s),{},{name:e.target.value}))},className:"add-call__input",placeholder:"Name",maxLength:30}),Object(d.jsx)("input",{type:"text",value:p,onChange:function(e){(""===e.target.value||k.test(e.target.value))&&(v(e.target.value),n(Object(o.a)(Object(o.a)({},s),{},{phone:e.target.value})))},className:"add-call__input",placeholder:"- 00XXX XXX XXX",maxLength:14}),Object(d.jsx)("input",{type:"time",value:C,onChange:function(e){g(e.target.value),n(Object(o.a)(Object(o.a)({},s),{},{time:e.target.value}))},className:"add-call__input"})]}),Object(d.jsx)(x,{click:function(){var e=!1,l=!1,a=!1;if(j.length>1&&j.length<=30&&(e=!0),14===p.length&&(l=!0),(void 0!==C||C)&&(a=!0),e&&l&&a){var c=function(e,t){var l=new Date(t),a=l.getHours(),c=l.getMinutes(),s=l.getSeconds(),n=l-i(a,c,s);function i(e,t,l){return 36e5*e+6e4*t+1e3*l}var r=e.split("");return n+i("0"===r[0]&&"0"===r[1]?0:"0"===r[0]&&"0"!==r[1]?r[1]:r[0]+r[1],"0"===r[3]&&"0"===r[4]?0:"0"===r[3]&&"0"!==r[4]?r[4]:r[3]+r[4],0)}(C,Date.now()),i=!1;Date.now()>c&&(i=!0);var r=Object(o.a)(Object(o.a)({id:Date.now()},s),{},{finish:i,milisec:c});t(r),S(j,p,C,c,i),b(""),v(""),g(""),n({})}else alert("Enter all form fields")},type:"button",classes:"add-call__button",text:"Add call"})]})]})},p=(l(20),function(e){var t=!1;e.call.finish&&(t=!0);var l=e.call.hidden?"hidden":"",a="calls-list__item ".concat(l);return Object(d.jsxs)("div",{className:a,children:[Object(d.jsx)("div",{className:"calls-list__col name-col",children:e.call.name}),Object(d.jsx)("div",{className:"calls-list__col phone-col",children:e.call.phone}),Object(d.jsx)("div",{className:"calls-list__col time-col","data-milisec":e.call.milisec,children:e.call.time}),Object(d.jsx)("div",{className:"calls-list__col time-delete",children:Object(d.jsx)("span",{onClick:function(){return e.remove(e.call)},children:"delete"})}),Object(d.jsx)("div",{className:"calls-list__col time-finish",children:Object(d.jsx)("input",{checked:t,type:"checkbox",disabled:!0})})]})}),v=function(){return Object(d.jsxs)("div",{className:"calls-list__buttons-wrap",children:[Object(d.jsx)(x,{type:"button",classes:"calls-list__button",text:"All"}),Object(d.jsx)(x,{type:"button",classes:"calls-list__button",text:"Next"}),Object(d.jsx)(x,{type:"button",classes:"calls-list__button",text:"Finished"})]})},N=function(e){var t=e.calls,l=e.remove,c=e.sortTimeAsc,s=e.sortTimeDesc,n=Object(a.useState)("all"),i=Object(r.a)(n,2);i[0],i[1];return Object(d.jsxs)("div",{className:"calls-list",children:[Object(d.jsxs)("div",{className:"calls-list__table",children:[Object(d.jsxs)("div",{className:"calls-list__headers",children:[Object(d.jsxs)("div",{className:"calls-list__col name-col",children:[Object(d.jsx)("span",{children:"Name"}),Object(d.jsxs)("div",{className:"calls-list__sort-block",children:[Object(d.jsx)("div",{className:"calls-list__sort-block-up"}),Object(d.jsx)("div",{className:"calls-list__sort-block-down"})]})]}),Object(d.jsx)("div",{className:"calls-list__col phone-col",children:"Phone number"}),Object(d.jsxs)("div",{className:"calls-list__col time-col",children:[Object(d.jsx)("span",{children:"Time"}),Object(d.jsxs)("div",{className:"calls-list__sort-block",children:[Object(d.jsx)("div",{className:"calls-list__sort-block-up",onClick:c}),Object(d.jsx)("div",{className:"calls-list__sort-block-down",onClick:s})]})]}),Object(d.jsx)("div",{className:"calls-list__col time-delete"}),Object(d.jsx)("div",{className:"calls-list__col time-finish"})]}),Object(d.jsx)("div",{className:"calls-list__cols-wrap",children:t.length?t.map((function(e){return Object(d.jsx)(p,{remove:l,call:e},e.milisec)})):Object(d.jsx)("div",{className:"calls-list__item",children:Object(d.jsx)("h1",{className:"calls-list__no-calls",children:"no calls today"})})})]}),Object(d.jsx)(v,{})]})},f=l(7),C=l.n(f),g=function(e){return Object(d.jsx)("header",{className:C.a.header,children:Object(d.jsx)("div",{className:"".concat(C.a.header__container,"  container"),children:Object(d.jsx)(u,{classes:"".concat(C.a.header__title," main-title"),text:"Today calls app"})})})},S=l(3),k=l.n(S),w=function(e){return Object(d.jsx)("div",{className:k.a.nextCall,children:Object(d.jsxs)("div",{className:k.a.nextCall__wrap,children:[Object(d.jsx)(u,{text:"Next call"}),Object(d.jsxs)("div",{className:k.a.nextCall__inputWrap,children:[Object(d.jsx)("div",{className:k.a.nextCall__input+" "+k.a.nextCall__inputName,children:"User 1"}),Object(d.jsx)("div",{className:k.a.nextCall__input+" "+k.a.nextCall__inputPhone,children:"032984234234"}),Object(d.jsx)("div",{className:k.a.nextCall__input+" "+k.a.nextCall__inputTime,children:" 4:20"})]})]})})};var y=function(){var e=O(),t=e.getItemsFromStorage,l=e.deleteItemsFromStorage,c=Object(a.useState)((function(){return t()})),s=Object(r.a)(c,2),n=s[0],o=s[1],_=Object(a.useState)("all"),j=Object(r.a)(_,2);return j[0],j[1],Object(d.jsxs)("div",{className:"App",children:[Object(d.jsx)(g,{}),Object(d.jsx)("div",{className:"container",children:Object(d.jsxs)("div",{className:"main-wrap",children:[Object(d.jsx)(w,{}),Object(d.jsxs)("div",{className:"calls-wrap",children:[Object(d.jsx)(h,{create:function(e){o([].concat(Object(i.a)(n),[e]))}}),Object(d.jsx)(N,{remove:function(e){o(n.filter((function(t){return t.id!==e.id}))),l(e.milisec)},calls:n,sortTimeAsc:function(){o((function(e){return e.sort((function(e,t){return e.milisec-t.milisec}))}))},sortTimeDesc:function(){o((function(e){return e.sort((function(e,t){return t.milisec-e.milisec}))}))}})]})]})})]})};n.a.render(Object(d.jsx)(c.a.StrictMode,{children:Object(d.jsx)(y,{})}),document.getElementById("root"))}],[[21,1,2]]]);
//# sourceMappingURL=main.80c951b0.chunk.js.map