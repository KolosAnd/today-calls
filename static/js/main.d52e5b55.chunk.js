(this["webpackJsonptoday-calls-app"]=this["webpackJsonptoday-calls-app"]||[]).push([[0],{175:function(e,t,a){},176:function(e,t,a){"use strict";a.r(t);var l=a(1),n=a.n(l),c=a(63),s=a.n(c),i=a(2),r=a(3),o=(a(73),a(6)),d=a(38),u=a.n(d),_=a(0),j=function(e){return Object(_.jsx)("div",{className:u.a.title,children:Object(_.jsx)("h2",{className:"".concat(u.a.title__name," ").concat(e.classes),children:e.text})})},m=(a(40),function(e,t){if(void 0!=e){var a=new Date(t),l=a.getHours(),n=a.getMinutes(),c=a.getSeconds(),s=a-b(l,n,c),i=10*e[0]+Number(e[1]),r=10*e[3]+Number(e[4]);return"number"==typeof e?s+e:s+b(i,r,0)}});function b(e,t,a){return 36e5*e+6e4*t+1e3*a}var x=function(e){if("string"!==typeof e){if(!e.toString())throw new Error("Invalid input");e=e.toString()}return b(10*e[0]+Number(e[1]),10*e[3]+Number(e[4]),0)},O=a(20),h=a(12),p=a(68),v=Object(l.createContext)(null),f={id:0,name:"",phone:"",time:"",milisec:0},N=O.a({name:O.b().required("\u041e\u0431\u044f\u0437\u0430\u0442\u0435\u043b\u044c\u043d\u043e\u0435 \u043f\u043e\u043b\u0435").test("\u0422\u043e\u043b\u044c\u043a\u043e \u0442\u0435\u043a\u0441\u0442","\u0422\u043e\u043b\u044c\u043a\u043e \u0442\u0435\u043a\u0441\u0442",(function(e){return/^\D+$/.test(e)})).min(2,"\u041c\u0438\u043d\u0438\u043c\u0443\u043c 2 \u0441\u0438\u043c\u0432\u043e\u043b\u0430").max(30,"\u041c\u0430\u043a\u0441\u0438\u043c\u0443\u043c 30 \u0441\u0438\u043c\u0432\u043e\u043b\u043e\u0432").typeError("\u0422\u043e\u043b\u044c\u043a\u043e \u0442\u0435\u043a\u0441\u0442"),phone:O.b().required("\u041e\u0431\u044f\u0437\u0430\u0442\u0435\u043b\u044c\u043d\u043e\u0435 \u043f\u043e\u043b\u0435").test("+(xxx)-xxx-xxx-xxx","+(xxx)-xxx-xxx-xxx",(function(e){return/^\+|(0{2})|[\(]?\d{3}[\)]?[-\s]?\d{3}[-\s]?\d{3}[-\s]?\d{3}$/.test(e)})).min(13,"\u0422\u0435\u043b\u0435\u0444\u043e\u043d \u043c\u0438\u043d\u0438\u043c\u0443\u043c 13 \u0441\u0438\u043c\u0432\u043e\u043b\u043e\u0432").max(18,"\u0422\u0435\u043b\u0435\u0444\u043e\u043d \u043c\u0430\u043a\u0441\u0438\u043c\u0443\u043c 18 \u0441\u0438\u043c\u0432\u043e\u043b\u043e\u0432").typeError("\u0412\u0432\u0435\u0434\u0438\u0442\u0435 \u043a\u043e\u0440\u0435\u043a\u0442\u043d\u044b\u0439 \u043d\u043e\u043c\u0435\u0440"),time:O.b().required("\u041e\u0431\u044f\u0437\u0430\u0442\u0435\u043b\u044c\u043d\u043e\u0435 \u043f\u043e\u043b\u0435").test("\u041d\u0435\u043a\u043e\u0440\u0435\u043a\u0442\u043d\u0430\u044f \u0434\u0430\u0442\u0430","\u041d\u0435\u043a\u043e\u0440\u0435\u043a\u0442\u043d\u0430\u044f \u0434\u0430\u0442\u0430",(function(e){return/^\d\d:\d\d+$/.test(e)})).typeError("\u0422\u043e\u043b\u044c\u043a\u043e \u0434\u0430\u0442\u0430")}).required(),C=function(e){var t,a,n,c=e.addOneCall,s=Object(l.useState)(f),i=Object(r.a)(s,2),d=i[0],u=i[1],b=Object(l.useContext)(v).calls,x={resolver:Object(p.a)(N)},O=Object(h.d)(x),C=O.register,g=O.formState.errors,k=O.handleSubmit;O.reset;return Object(l.useEffect)((function(){if(d.id){var e=b.concat([d]);localStorage.setItem("list",JSON.stringify(e)),c(d),u(f)}}),[d]),Object(_.jsxs)("div",{className:"add-call",children:[Object(_.jsx)(j,{text:"Add call"}),Object(_.jsxs)("form",{onSubmit:k((function(e){var t=new FormData;t.append("name",e.name),t.append("phone",e.phone),t.append("time",e.time);var a=m(d.time,Date.now()),l={id:Date.now(),milisec:a,phone:"00"+d.phone.replaceAll(/[\(\)\-\+\s]/g,"").replaceAll(/^(0{2})/g,"")};u(Object(o.a)(Object(o.a)({},d),l))})),children:[Object(_.jsxs)("div",{className:"add-call__inputs-wrap",children:[Object(_.jsxs)("div",{className:"add-call__input-block name",children:[Object(_.jsx)("input",Object(o.a)(Object(o.a)({},C("name")),{},{value:d.name,onChange:function(e){u(Object(o.a)(Object(o.a)({},d),{},{name:e.target.value}))},name:"name",type:"text",className:g.name?"add-call__input error":"add-call__input",placeholder:"Name",maxLength:30})),Object(_.jsx)("span",{className:"add-call__error-message",children:null===(t=g.name)||void 0===t?void 0:t.message})]}),Object(_.jsxs)("div",{className:"add-call__input-block phone",children:[Object(_.jsx)("input",Object(o.a)(Object(o.a)({},C("phone")),{},{onChange:function(e){u(Object(o.a)(Object(o.a)({},d),{},{phone:e.target.value}))},name:"phone",value:d.phone,placeholder:"+(\u0425\u0425X)-XX\u0425-X\u0425\u0425-\u0425\u0425\u0425",type:"text",className:g.phone?"add-call__input error":"add-call__input"})),Object(_.jsx)("span",{className:"add-call__error-message",children:null===(a=g.phone)||void 0===a?void 0:a.message})]}),Object(_.jsxs)("div",{className:"add-call__input-block time",children:[Object(_.jsx)("input",Object(o.a)(Object(o.a)({type:"time",name:"time"},C("time")),{},{value:d.time,onChange:function(e){u(Object(o.a)(Object(o.a)({},d),{},{time:e.target.value}))},className:g.time?"add-call__input error":"add-call__input"})),Object(_.jsx)("span",{className:"add-call__error-message",children:null===(n=g.time)||void 0===n?void 0:n.message})]})]}),Object(_.jsx)("button",{type:"submit",className:"button add-call__button",children:"Add call"})]})]})},g=(a(175),function(e){var t=e.remove,a=e.call,n=e.hidden,c=Object(l.useState)(!1),s=Object(r.a)(c,2),i=s[0],o=s[1],d=Object(l.useState)("calls-list__item"),u=Object(r.a)(d,2),j=u[0],b=u[1];return Object(l.useEffect)((function(){if(a.time){var e=x(a.time),t=m(e,Date.now());Date.now()>t&&o(!0)}}),[a]),Object(l.useEffect)((function(){"next"==n&&b(i?" hidden":""),"finished"==n&&b(i?"":" hidden"),"all"==n&&b("")}),[n]),Object(_.jsx)(_.Fragment,{children:a&&Object(_.jsxs)("div",{className:"calls-list__item ".concat(j),children:[Object(_.jsx)("div",{className:"calls-list__col name-col",children:a.name}),Object(_.jsx)("div",{className:"calls-list__col phone-col",children:a.phone}),Object(_.jsx)("div",{className:"calls-list__col time-col","data-milisec":a.milisec,children:a.time}),Object(_.jsx)("div",{className:"calls-list__col time-delete",children:Object(_.jsx)("span",{onClick:function(){return t(a.id)},children:"delete"})}),Object(_.jsx)("div",{className:"calls-list__col time-finish",children:Object(_.jsx)("input",{checked:i,type:"checkbox",disabled:!0})})]})})}),k=function(e){var t=e.calls,a=e.remove,n=Object(l.useState)(""),c=Object(r.a)(n,2),s=c[0],i=c[1],o=Object(l.useState)("all"),d=Object(r.a)(o,2),u=d[0],j=d[1];return Object(_.jsxs)("div",{className:"calls-list",children:[Object(_.jsxs)("div",{className:"calls-list__table",children:[Object(_.jsxs)("div",{className:"calls-list__headers",children:[Object(_.jsxs)("div",{className:"calls-list__col name-col",children:[Object(_.jsx)("span",{children:"Name"}),Object(_.jsxs)("div",{className:"calls-list__sort-block",children:[Object(_.jsx)("div",{className:"calls-list__sort-block-up",onClick:function(){return i("ascName")}}),Object(_.jsx)("div",{className:"calls-list__sort-block-down",onClick:function(){return i("descName")}})]})]}),Object(_.jsx)("div",{className:"calls-list__col phone-col",children:"Phone number"}),Object(_.jsxs)("div",{className:"calls-list__col time-col",children:[Object(_.jsx)("span",{children:"Time"}),Object(_.jsxs)("div",{className:"calls-list__sort-block",children:[Object(_.jsx)("div",{className:"calls-list__sort-block-up",onClick:function(){return i("ascTime")}}),Object(_.jsx)("div",{className:"calls-list__sort-block-down",onClick:function(){return i("descTime")}})]})]}),Object(_.jsx)("div",{className:"calls-list__col time-delete"}),Object(_.jsx)("div",{className:"calls-list__col time-finish"})]}),Object(_.jsx)("div",{className:"calls-list__cols-wrap",children:t.length?t.sort((function(e,t){return"ascTime"===s?function(e,t){return e.milisec-t.milisec}(e,t):"descTime"===s?function(e,t){return t.milisec-e.milisec}(e,t):"ascName"===s?function(e,t){return e.name>t.name?1:e.name<t.name?-1:0}(e,t):"descName"===s?function(e,t){return e.name<t.name?1:e.name>t.name?-1:0}(e,t):void 0})).map((function(e){return Object(_.jsx)(g,{hidden:u,remove:a,call:e},e.id)})):Object(_.jsx)("div",{className:"calls-list__item",children:Object(_.jsx)("h1",{className:"calls-list__no-calls",children:"no calls today"})})})]}),Object(_.jsxs)("div",{className:"calls-list__buttons-wrap",children:[Object(_.jsx)("button",{onClick:function(){return j("all")},type:"button",className:"button calls-list__button",children:"All"}),Object(_.jsx)("button",{onClick:function(){return j("next")},type:"button",className:"button calls-list__button",children:"Next"}),Object(_.jsx)("button",{onClick:function(){return j("finished")},type:"button",className:"button calls-list__button",children:"Finished"})]})]})},w=a(28),S=a.n(w),y=function(e){return Object(_.jsx)("header",{className:S.a.header,children:Object(_.jsx)("div",{className:"".concat(S.a.header__container,"  container"),children:Object(_.jsx)(j,{classes:"".concat(S.a.header__title," main-title"),text:"Today calls app"})})})},T=a(7),D=a.n(T),E=function(e){var t=e.calls,a=Object(l.useState)([]),n=Object(r.a)(a,2),c=n[0],s=n[1],i=Object(l.useState)({}),o=Object(r.a)(i,2),d=o[0],u=o[1];Object(l.useEffect)((function(){t.length>0&&s(t.sort((function(e,t){return e.milisec>t.milisec?1:-1})))}),[t]),Object(l.useEffect)((function(){if(c.length>0){for(var e=0;e<c.length;e++){var t=x(c[e].time),a=m(t,Date.now());if(Date.now()<a){u(c[e]);break}u(null)}var l=b();return function(){clearInterval(l)}}}),[c]);var b=function(){return setInterval((function(){for(var e=0;e<c.length;e++){var t=x(c[e].time),a=m(t,Date.now());if(Date.now()<a){u(c[e]);break}}}),6e4)};return Object(_.jsx)("div",{className:D.a.nextCall,children:Object(_.jsxs)("div",{className:D.a.nextCall__wrap,children:[Object(_.jsx)(j,{text:"Next call"}),d?Object(_.jsxs)("div",{className:D.a.nextCall__inputWrap,children:[Object(_.jsx)("div",{className:D.a.nextCall__input+" "+D.a.nextCall__inputName,children:d.name}),Object(_.jsx)("div",{className:D.a.nextCall__input+" "+D.a.nextCall__inputPhone,children:d.phone}),Object(_.jsxs)("div",{className:D.a.nextCall__input+" "+D.a.nextCall__inputTime,children:[" ",d.time]})]}):Object(_.jsx)("div",{className:D.a.nextCall__inputWrap,children:Object(_.jsx)("div",{className:D.a.nextCall__input+" "+D.a.nextCall__inputName,children:"No next call"})})]})})},I=a(37),A="filter/ALL",W="filter/NEXT",J="filter/FINISH";var H=function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:{value:"all"},t=arguments.length>1?arguments[1]:void 0;switch(t.type){case A:return{value:"all"};case W:return{value:"next"};case J:return{value:"finish"};default:return e}},P=a(67);var X=function(){var e=Object(I.createStore)(H,Object(P.composeWithDevTools)());e.subscribe((function(){return console.log(e.getState())}));var t=Object(l.useState)((function(){return function(){var e=JSON.parse(localStorage.getItem("list"));return null==e?[]:e}()})),a=Object(r.a)(t,2),n=a[0],c=a[1];return Object(l.useEffect)((function(){n.length&&localStorage.setItem("list",JSON.stringify(n))}),[n]),Object(_.jsx)(v.Provider,{value:{calls:n},children:Object(_.jsxs)("div",{className:"App",children:[Object(_.jsx)(y,{}),Object(_.jsx)("div",{className:"container",children:Object(_.jsxs)("div",{className:"main-wrap",children:[Object(_.jsx)(E,{calls:n}),Object(_.jsxs)("div",{className:"calls-wrap",children:[Object(_.jsx)(C,{addOneCall:function(e){c([].concat(Object(i.a)(n),[e]))}}),Object(_.jsx)(k,{remove:function(e){var t=n.filter((function(t){return t.id!==e}));c(t)},calls:n})]})]})})]})})};s.a.render(Object(_.jsx)(n.a.StrictMode,{children:Object(_.jsx)(X,{})}),document.getElementById("root"))},28:function(e,t,a){e.exports={header:"Header_header__2vzj5",header__container:"Header_header__container__3VRL9",header__title:"Header_header__title__2pv23"}},38:function(e,t,a){e.exports={title:"Title_title__38Rue",title__name:"Title_title__name__G7AnC","main-title":"Title_main-title__15ozn"}},40:function(e,t,a){},7:function(e,t,a){e.exports={nextCall:"NextCall_nextCall__2hiNO",nextCall__wrap:"NextCall_nextCall__wrap__E-JmG",nextCall__title:"NextCall_nextCall__title__3uk50",nextCall__inputWrap:"NextCall_nextCall__inputWrap__3k8d4",nextCall__inputsWrap:"NextCall_nextCall__inputsWrap__2i2Go",nextCall__input:"NextCall_nextCall__input__3VO86",nextCall__inputName:"NextCall_nextCall__inputName__3dOlZ",nextCall__inputPhone:"NextCall_nextCall__inputPhone__37SD1",nextCall__inputTime:"NextCall_nextCall__inputTime__2rI6V"}},73:function(e,t,a){}},[[176,1,2]]]);
//# sourceMappingURL=main.d52e5b55.chunk.js.map