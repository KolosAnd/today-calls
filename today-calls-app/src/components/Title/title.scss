.title {

  &__name {
    color: #fff;
    margin: 0 0 25px;
    font-size: 25px;
    text-decoration: none;
    line-height: 25px;
  }
}