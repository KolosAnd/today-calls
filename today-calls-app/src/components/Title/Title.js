import "./title.scss"
import React from "react";

export const Title = (props) => {
    return (
        <div className="title">
            <h2 className="title__name">{props.text}</h2>
        </div>
    )
}

