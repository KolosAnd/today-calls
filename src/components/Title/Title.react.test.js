import React from 'react';
import renderer from 'react-test-renderer';
import {Title} from "./Title.js";

describe('Title component', () => {
    test('Title without props', () => {
        const component = renderer.create(<Title/>,);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Title with test props', () => {
        const component = renderer.create(<Title classes={""} text={"Next call"}/>,);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})