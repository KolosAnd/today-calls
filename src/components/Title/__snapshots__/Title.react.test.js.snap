// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title changes the class when hovered 1`] = `
<div
  className="title"
>
  <h2
    className="title__name undefined"
  />
</div>
`;

exports[`Title component Title with test props 1`] = `
<div
  className="title"
>
  <h2
    className="title__name "
  >
    Next call
  </h2>
</div>
`;

exports[`Title component Title without props 1`] = `
<div
  className="title"
>
  <h2
    className="title__name undefined"
  />
</div>
`;

exports[`Title with test props 1`] = `
<div
  className="title"
>
  <h2
    className="title__name "
  >
    Next call
  </h2>
</div>
`;

exports[`Title without props 1`] = `
<div
  className="title"
>
  <h2
    className="title__name undefined"
  />
</div>
`;
